cmd/noe/main.go

```diff
@@ -22,21 +24,43 @@ import (
 )
 
 func main() {
-	var preferredArch, systemOS string
+	var preferredArch, schedulableArchs, systemOS string
 	var metricsAddr string
 	var registryProxies, matchNodeLabels string
 
 	flag.StringVar(&preferredArch, "preferred-arch", "amd64", "Preferred architecture when placing pods")
+	flag.StringVar(&schedulableArchs, "cluster-schedulable-archs", "", "Comma separated list of architectures schedulable in the cluster")
 	flag.StringVar(&systemOS, "system-os", "linux", "Sole OS supported by the system")
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&metricsAddr, "registry-proxies", "", "Proxies to substitute in the registry URL in the form of docker.io=docker-proxy.company.corp,quay.io=quay-proxy.company.corp")
 	flag.StringVar(&matchNodeLabels, "match-node-labels", "", "A set of pod label keys to match against node labels in the form of key1,key2")
 	flag.Parse()
 
-	Main(signals.SetupSignalHandler(), "./", preferredArch, systemOS, metricsAddr, registryProxies, matchNodeLabels)
+	Main(signals.SetupSignalHandler(), "./", preferredArch, schedulableArchs, systemOS, metricsAddr, registryProxies, matchNodeLabels)
 }
 
-func Main(ctx context.Context, certDir, preferredArch, systemOS, metricsAddr, registryProxies, matchNodeLabels string) {
+func Main(ctx context.Context, certDir, preferredArch, schedulableArchs, systemOS, metricsAddr, registryProxies, matchNodeLabels string) {
+	var schedulableArchSlice []string
+
+	if schedulableArchs != "" {
+		schedulableArchSlice = strings.Split(schedulableArchs, ",")
+	}
+
+	if preferredArch != "" && schedulableArchs != "" {
+		supported := false
+		for _, schedulableArch := range schedulableArchSlice {
```

Comment:
don't want to sound overkill but https://pkg.go.dev/slices I see in the repo we are already building the code with 1.21 so we may as well try and use the slices.Contains for simplicity

Reply:
Or simply use 	"golang.org/x/exp/slices"
 and then, in 1.21 port to the standard library

Reply:
nice! Adopted it!